src/frontend/js/components/DashBoard/Router.spec.tsx

```diff
@@ -0,0 +1,154 @@
+import createQueryClient from '../../utils/react-query/createQueryClient';
+import { location } from '../../utils/indirection/window';
+import { DashBoardRoute } from './routes';
+import ProtectedRoute from './ProtectedRoute';
+
+jest.mock('utils/indirection/window', () => ({
+  location: {
+    assign: jest.fn(),
+  },
+}));
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockFactories
+    .ContextFactory({
+      authentication: { backend: 'fonzie', endpoint: 'https://demo.endpoint' },
+      joanie_backend: { endpoint: 'https://joanie.endpoint' },
+    })
+    .generate(),
+}));
+
+describe('<DashBoardRouter />', () => {
+  const Wrapper = ({
+    initial_entries,
+    routes,
+  }: {
+    initial_entries: Array<string>;
+    routes?: Array<DashBoardRoute>;
+  }) => {
+    if (!routes) {
+      routes = [
+        {
+          path: '/courses',
+          title: 'courses',
+          element: <h2>Courses</h2>,
+        },
+        {
+          path: '/preferences',
+          title: 'preferences',
+          element: <h2>Preferences</h2>,
+        },
+        {
+          path: '/protected_path',
+          title: 'protected',
+          element: (
+            <ProtectedRoute isAllowed={false} redirectPath="/">
+              <h2>Never ever</h2>
+            </ProtectedRoute>
+          ),
+        },
+        {
+          element: <h2>404 Not Found</h2>,
+          path: '*',
+        },
+      ];
+    }
+    return (
+      <QueryClientProvider client={createQueryClient({ persistor: true })}>
+        <IntlProvider locale="en">
+          <SessionProvider>
+            <MemoryRouter initialEntries={initial_entries}>
+              <DashBoardRouter routes={routes} />
+            </MemoryRouter>
+          </SessionProvider>
+        </IntlProvider>
+      </QueryClientProvider>
+    );
+  };
+
+  const initializeUser = (loggedin = true) => {
+    const user = loggedin ? FonzieUserFactory.generate() : null;
+
+    sessionStorage.setItem(
+      REACT_QUERY_SETTINGS.cacheStorage.key,
+      JSON.stringify(
+        PersistedClientFactory({ queries: [QueryStateFactory('user', { data: user })] }),
+      ),
+    );
+
+    if (loggedin) {
+      sessionStorage.setItem(RICHIE_USER_TOKEN, user.access_token);
+    }
+
+    return user;
+  };
+
+  beforeEach(() => {
+    sessionStorage.clear();
+  });
+
+  it('shows a navigation displaying all provided paths', async () => {
+    await act(async () => {
+      render(<Wrapper initial_entries={['/']} />);
+    });
+
+    const paths = ['/courses', '/preferences'];
+    const links = screen.getAllByRole('link');
+    let hrefs: Array<string> = [];
+
+    hrefs = links.reduce((res: Array<string>, item) => {
+      if (item.hasAttribute('href') && item.getAttribute('href'))
+        res.push(item.getAttribute('href') || '');
+      return res;
+    }, []);
+
+    paths.forEach((path) => {
+      expect(hrefs.includes(path)).toEqual(true);
+    });
```

Comment:
As I recommend in my comment above, you could store default routes into a `DEFAULT_ROUTES` variables. In this way, you will be able to simplify this check.

```suggestion
    DEFAULT_ROUTES.forEach(({path, title}) => {
      const $link = screen.getByRole('link', {name: title});
      expect($link.getAttribute('href')).toBe(path);
    });
```